### DISCUSSION TITLE
Feature Request: Add hyperbolic functions

### DISCUSSION THREAD

**Discussion by JayjeetAtGithub** [duckdb/duckdb#6600] [Ideas]
I need to use DuckDB for processing of High Energy Physics datasets. My queries use standard hyperbolic functions such as `cosh`. One example query can be found [here](https://github.com/RumbleDB/iris-hep-benchmark-athena/blob/master/queries/query-5/query.sql). I am planning to raise a PR with `cosh` implementation using `std::cosh` and If that looks good to the community, I can do the same for other hyperbolic functions.

**Comment by carlopi** [comment]
Cool project, do share when you are a bit further!

For this I would probably recommend to get started with a DuckDB extension, since has the advantage you can move faster (and break things...) AND you get to implement things with more flexibility.

Extension could then be either merged with main-line duckdb (without having to worry with conficts and all) or separately mainteined in a specialized extension.

The basic can be found in the [extension-template repo](https://github.com/duckdb/extension-template), then in your case it would be a matter of implementing the various trigonometric functions.

I would also recommend joining https://discord.duckdb.org, where you could get feedback / share your work.

**Reply by JayjeetAtGithub** [reply]
I see. Thanks a lot for the pointers. Let me take a look.
